stubs/reportlab/reportlab/graphics/shapes.pyi

```diff
@@ -1,11 +1,96 @@
-from _typeshed import Incomplete
-from typing import Final
-
-from reportlab.lib.attrmap import *
-from reportlab.lib.validators import *
+from _typeshed import Incomplete, SupportsItems
+from abc import abstractmethod
+from collections.abc import Iterable, Sequence
+from typing import Any, Final, Literal, NoReturn, TypedDict
+from typing_extensions import Self, TypeAlias, Unpack
+
+from reportlab.lib.colors import Color
+from reportlab.lib.validators import NoneOr, Validator
+from reportlab.pdfgen.canvas import Canvas
 from reportlab.platypus import Flowable
-
-from .transform import *
+from reportlab.platypus.flowables import _HAlignment, _VAlignment
+
+_IntBool: TypeAlias = Literal[0, 1]
+_AnyBool: TypeAlias = _IntBool | bool
```

Comment:
Nit: I would avoid using "Any" as part of the name, as this implies a "type escape hatch". Maybe just `_BoolOrInt` or `_Boolish`, `_BoolLike`?

Also, we usually just use `bool` in cases like this, as users should switch to using bools. Although I don't know the specifics of reportlab and whether using ints is common, or whether it's just left over from times when Python didn't have a bool type (like in some areas of the stdlib). Also, in cases where the default is `0`, using this type alias is probably easier. Your decision what's best here.

Reply:
I haven't fully decided yet what to do with these yet, since there appear to be some places where it may actually need to be a `int` rather than a `bool`, although I haven't verified that yet. It helps that `bool` is a subclass of `int` and `hash(True) == hash(1)` so the values are interchangeable in lookup tables.

But as you've pointed out there are a ton of places where the default value is either `1` or `0`, so it might be easier and more consistent to just use this alias everywhere. Let's leave it for now until I actually decide to clean up all these boolean arguments, I'm sure there are a lot of them that are just annotated as `int` currently because of stubgen.